README
Fretsboard
==========

[![Build Status](https://travis-ci.org/Linkid/fretsboard.svg?branch=master)](https://travis-ci.org/Linkid/fretsboard)

Fretsboard is a web scoreboard for the rythm game Frets On Fire.

Instance (old): http://www.prison.net/worldcharts/charts/


How to run
==========

On a virtualenv:

```bash
git clone git@github.com:Linkid/fretsboard.git
cd fretsboard
pip install -r requirements.txt
./manage.py migrate
./manage.py runserver
```


Origin
======

This application comes from the old [Frets On Fire Charts Server](https://sourceforge.net/projects/fretsonfire/files/fretsonfire-chartserver/).

Other scoreboard were developped, in PHP:
- fofchart (2008): https://sourceforge.net/projects/fofchart/
- fofcs (2008): https://sourceforge.net/projects/fofcs/
- fretsweb-php (2009): https://sourceforge.net/projects/fretsweb/

For more information, see the [Frets On Fire forum post](http://www.fretsonfire.net/forums/viewtopic.php?f=21&t=28485).


CHANGELOG.md
Release notes
=============

1.6 (unreleased)
----------------

1.5 (2021-04-29)
----------------

- build: add Python 3.9 compatibility
- ci: use Github Actions to run tests
- dependencies: update Django to 3.2
- doc: update the Travis link


1.4 (2020-08-10)
----------------

- Admin: display and filter verified songs
- Dependencies: update Django to 3.1
- Dependencies: update django-bootstrap3 to 12.0
- Dependencies: update Python to 3.8
- Dependencies: specify the minimal required Python version
- Templates: get translated difficulty in the song template
- Tests: add python 3.7 and 3.8 compatibility
- Translations: use lazy translations in models to get translated text in templates


1.3 (2019-12-05)
----------------

- Dependencies: update Cerealizer to 0.8.3
- Dependencies: update Django to 3.0
- Dependencies: update Python to 3.7
- Templates: add error pages (400, 403, 404 and 500)
- Translations: update languages.


1.2 (2018-10-21)
----------------

- Dependencies: update Cerealizer to 0.8.2
- Dependencies: update Django to 2.1


1.1 (2018-06-16)
----------------

- Dependencies: migrate to Django 2.0
- Dependencies: use the Cerealizer binary
- Doc: add a screenshot of the scoreboard
- Doc: add release notes
- Logs: add some logs
- Theme: add some links to the footer
- Theme: reduce the size of medals
- Translations: update languages.
- Upload: duplicate songs if needed
- Upload: filter songs by case-insensitive title


1.0 (2018-04-01)
----------------

The scoreboard has been rewritten.

- Django: migrate the scoreboard to Django 1.11
- Theme: use Bootstrap 3
- Theme: use a simple theme (white, not red)
- Theme: use a simple boostrap theme for the user page
- Translations: 2 languages are available (English, French).


fretsboard/settings.py
"""Django settings for fretsboard project."""

from pathlib import Path

# Build paths inside the project like this: BASE_DIR / 'subdir'.
BASE_DIR = Path(__file__).resolve().parent.parent


# Quick-start development settings - unsuitable for production
# See https://docs.djangoproject.com/en/1.11/howto/deployment/checklist/

# SECURITY WARNING: keep the secret key used in production secret!
SECRET_KEY = 'key'

# SECURITY WARNING: don't run with debug turned on in production!
DEBUG = True

ALLOWED_HOSTS = []


# Application definition

INSTALLED_APPS = [
    'django.contrib.admin',
    'django.contrib.auth',
    'django.contrib.contenttypes',
    'django.contrib.sessions',
    'django.contrib.messages',
    'django.contrib.staticfiles',

    'bootstrap3',
    'scoreboard',
]

MIDDLEWARE = [
    'django.middleware.security.SecurityMiddleware',
    'django.contrib.sessions.middleware.SessionMiddleware',
    'django.middleware.locale.LocaleMiddleware',
    'django.middleware.common.CommonMiddleware',
    'django.middleware.csrf.CsrfViewMiddleware',
    'django.contrib.auth.middleware.AuthenticationMiddleware',
    'django.contrib.messages.middleware.MessageMiddleware',
    'django.middleware.clickjacking.XFrameOptionsMiddleware',
]

ROOT_URLCONF = 'fretsboard.urls'

TEMPLATES = [
    {
        'BACKEND': 'django.template.backends.django.DjangoTemplates',
        'DIRS': [
            BASE_DIR / 'templates',
        ],
        'APP_DIRS': True,
        'OPTIONS': {
            'context_processors': [
                'django.template.context_processors.debug',
                'django.template.context_processors.request',
                'django.contrib.auth.context_processors.auth',
                'django.contrib.messages.context_processors.messages',
                'fretsboard.context_processors.piwik_code',
            ],
        },
    },
]

WSGI_APPLICATION = 'fretsboard.wsgi.application'


# Database
# https://docs.djangoproject.com/en/1.11/ref/settings/#databases

DATABASES = {
    'default': {
        'ENGINE': 'django.db.backends.sqlite3',
        'NAME': BASE_DIR / 'db.sqlite3',
    }
}


# Password validation
# https://docs.djangoproject.com/en/1.11/ref/settings/#auth-password-validators

AUTH_PASSWORD_VALIDATORS = [
    {
        'NAME': 'django.contrib.auth.password_validation.UserAttributeSimilarityValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.MinimumLengthValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.CommonPasswordValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.NumericPasswordValidator',
    },
]


# Internationalization
# https://docs.djangoproject.com/en/1.11/topics/i18n/

LOCALE_PATHS = [
    BASE_DIR / 'locale',
]

LANGUAGE_CODE = 'en-us'

LANGUAGES = (
    ('en', 'English'),
    ('fr', 'French'),
)

TIME_ZONE = 'UTC'

USE_I18N = True

USE_L10N = True

USE_TZ = True

# Static files (CSS, JavaScript, Images)
# https://docs.djangoproject.com/en/1.11/howto/static-files/

STATIC_ROOT = BASE_DIR / 'static'
STATIC_URL = '/static/'
STATICFILES_DIRS = (
    BASE_DIR / 'fretsboard' / 'static',
)

# Default primary key field type
# https://docs.djangoproject.com/en/3.2/ref/settings/#default-auto-field

DEFAULT_AUTO_FIELD = 'django.db.models.BigAutoField'

# analytics
PIWIK_URL = ''
PIWIK_SITE_ID = ''
